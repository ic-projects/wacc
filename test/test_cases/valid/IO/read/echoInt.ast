-- Test: echoInt.wacc

-- Uploaded file: 
---------------------------------------------------------------
# echo the user's input int

# Output:
# enter an integer to echo
# #input#
# #output#

# Program:

begin
	int x = 1 ;
  println "enter an integer to echo";
	read x ; 
	println x 
end
---------------------------------------------------------------

-- Compiler Output:
-- Compiling...
-- Printing AST...
echoInt.ast contents are:
===========================================================
0	Program
1	  - int main()
2	    - DECLARE
3	      - TYPE
4	        - int
5	      - LHS
6	        - x
7	      - RHS
8	        - 1
9	    - PRINTLN
10	      - "enter an integer to echo"
11	    - READ
12	      - x
13	    - PRINTLN
14	      - x
===========================================================
-- Finished

