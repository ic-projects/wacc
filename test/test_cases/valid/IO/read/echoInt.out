-- Compiling...
-- Assembling and Linking...
-- Executing...
===========================================================
enter an integer to echo
1
===========================================================
The exit code is 0.
-- Finished
